src/java.base/share/classes/javax/net/ssl/SSLParameters.java

```diff
@@ -696,4 +698,76 @@ public void setApplicationProtocols(String[] protocols) {
         }
         applicationProtocols = tempProtocols;
     }
+
+    /**
+     * Returns a prioritized array of signature scheme names that can be used
+     * over the SSL/TLS/DTLS protocols.
+     * <p>
+     * The returned array includes {@code String} names from the list of
+     * standard signature schemes in the <a href=
+     * "{@docRoot}/../specs/security/standard-names.html#signature-schemes">
+     * Signature Schemes</a> section of the Java Cryptography
+     * Architecture Standard Algorithm Name Documentation, and may also
+     * include other signature schemes that the provider supports.
+     * <p>
+     * The array could be empty (zero-length), in which case the underlying
+     * provider-specific default signature schemes will be used.
+     *
+     * @implSpec
+     * This method will return a new array each time it is invoked.
+     *
+     * @return a non-null, possibly zero-length array of signature scheme
+     *         {@code String}s.  The array is placed in descending order of
+     *         signature scheme preference.
+     * @see #setSignatureSchemes
+     *
+     * @since 19
+     */
+    public String[] getSignatureSchemes() {
+        return signatureSchemes.clone();
+    }
+
+    /**
+     * Sets the prioritized array of signature scheme names that
+     * can be used over the SSL/TLS/DTLS protocols.
+     * <p>
+     * Note that the standard list of signature scheme names may be found in
+     * the <a href=
+     * "{@docRoot}/../specs/security/standard-names.html#signature-schemes">
+     * Signature Schemes</a> section of the Java Cryptography
+     * Architecture Standard Algorithm Name Documentation.  Providers
+     * may support signature schemes not found in this list or might not
+     * use the recommended name for a certain signature scheme.
+     *
+     * @implSpec
+     * This method will make a copy of the {@code signatureSchemes} array.
+     *
+     * @param signatureSchemes an ordered array of signature scheme names,
+     *        with the first entry being the most preferred. If the array
+     *        is empty (zero-length), the prodiver-specific default signature
+     *        schemes will be used for the SSL/TLS/DTLS connection.
+     * @throws IllegalArgumentException if signatureSchemes is null, or if
+     *        any element in a non-empty array is null or an
+     *        empty (zero-length) string.
+     *
+     * @see #getSignatureSchemes
+     *
+     * @since 19
+     */
+    public void setSignatureSchemes(String[] signatureSchemes) {
+        if (signatureSchemes == null) {
+            throw new IllegalArgumentException("signatureSchemes was null");
+        }
+
+        String[] tempSchemes = signatureSchemes.clone();
+        for (String scheme : tempSchemes) {
```

Comment:
If we want to reuse the result, we may have to cache something.  It is not good to me.  The parsing of the signature scheme names actually depends on the provider.  So at this point, in the Java SE API specification, it is not easy to know how to parse the strings to me.

Reply:
Yes you are right, it would require a setter where the provider can register the resolved array into the parameter instance (if we need to support multiple providers). But it would be a good optimization assuming clients keep the parameters around in a frequent Connection scenario.

Reply:
I see your point.  The API default implementation could be complicated, if considering loading and unloading providers at runtime.  I agree with you about the potential performance impact (minimal, but we may want to optimizing), and let's see if we could have a solution later.  Since it is a common issue, I will not address it in this PR.